bibliopixel/control/api.py

```diff
@@ -0,0 +1,56 @@
+from http.server import HTTPServer, BaseHTTPRequestHandler
+import urllib
+import getpass, platform, sys, threading
+from .. util import log
+from . control import ExtractedControl
+
+class Api(ExtractedControl):
+    EXTRACTOR = {
+        'keys_by_type': {
+            'direct_command': ['action'],
```

Comment:
Quibble: I would make these into tuples like `('action',)`, not lists - because tuples are immutable.